### PR TITLE
txn: support pessimistic rollback with scan first in mockstore

### PR DESCRIPTION
**Pull request by cfzjywxk**
### What problem does this PR solve?

Issue Number: ref #50731 

Problem Summary:
Introduce lock scan phase for pessimistic lock rollback.

### Check List

Tests 

Side effects

Documentation

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by zyguan — APPROVED** [review]

File: pkg/store/mockstore/unistore/tikv/mvcc.go [hunk 1]
```diff
@@ -1427,6 +1445,22 @@ func (store *MVCCStore) appendScannedLock(locks []*kvrpcpb.LockInfo, it *locksto
 	return locks
 }
 
+// ScanPessimisticLocks returns matching pessimistic locks.
+func (store *MVCCStore) ScanPessimisticLocks(reqCtx *requestCtx, startTS uint64, forUpdateTS uint64) ([]*kvrpcpb.LockInfo, error) {
```
Comment (MyonKeminta) [inline pkg/store/mockstore/unistore/tikv/mvcc.go:1449]: Is it better to be a private method, as it's only internally used by `PessimisticRollbackWithScanFirst`?

**Review by MyonKeminta — APPROVED** [review]
Hmmm it doesn't matter

**Pull request merged**